ca/client.go

```diff
@@ -605,6 +608,34 @@ retry:
 	return &version, nil
 }
 
+// Capabilities performs the capabilities request to the CA and returns the
+// api.Capabilities struct.
+func (c *Client) Capabilities() (*api.CapabilitiesResponse, error) {
+	var retried bool
+	u := c.endpoint.ResolveReference(&url.URL{Path: "/capabilities"})
+retry:
+	resp, err := c.client.Get(u.String())
+	if err != nil {
+		return nil, clientError(err)
+	}
+	switch {
+	case resp.StatusCode == http.StatusNotFound:
+		return nil, ErrNotFound
+	case resp.StatusCode >= 400:
+		if !retried && c.retryOnError(resp) {
+			retried = true
+			goto retry
+		}
+		return nil, readError(resp.Body)
+	default:
+		var capabilities api.CapabilitiesResponse
+		if err := readJSON(resp.Body, &capabilities); err != nil {
+			return nil, errs.Wrapf(http.StatusInternalServerError, err, "client.Capabilities; error reading %s", u)
```

Comment:
Personal preference; feel free to ignore.

```suggestion
			return nil, errs.Wrapf(http.StatusInternalServerError, err, "client.Capabilities; error reading %q", u)
```